manifests/init.pp

```diff
@@ -85,9 +87,24 @@
   Stdlib::Absolutepath $config_path_v6          = $bird::params::config_path_v6,
   Optional[Stdlib::Filesource] $config_file_v6  = undef,
   Optional[String[1]] $config_template_v6       = undef,
+  Boolean $manage_repository                    = false,
 ) inherits bird::params {
 
-  ensure_packages([$package_name_v4], {'ensure' => 'present'})
+  if $manage_repository {
+    yumrepo{'bird':
+      baseurl  => 'ftp://bird.network.cz/pub/bird/centos/7/x86_64/',
+      descr    => 'Official bird packages from CZ.NIC',
+      enabled  => 1,
+      gpgcheck => 1,
+      gpgkey   => 'ftp://bird.network.cz/pub/bird/centos/7/x86_64/RPM-GPG-KEY-network.cz',
+    }
+    # set a dependency to package resources
+    $dependency = Yumrepo['bird']
```

Comment:
Is it possibly to just declare the dependency here instead of passing something to `ensure_packages`.  Maybe otherwise it could conflict with any previous declaration of the package (which is what you're trying to cover by using `ensure_packages`)??
```puppet
Yumrepo['bird'] -> Package[$package_name_v4]
if $enable_v6 { Yumrepo['bird'] -> Package[$package_name_v6] }
```
or maybe
```puppet
Yumrepo['bird'] -> Package <| name ==  $package_name_v4 or name ==  $package_name_v6 |>
```

Reply:
or with the `before` metaparameter in the Yumrepo['bird'] declaration.
```puppet
before => Package <| name ==  $package_name_v4 or name ==  $package_name_v6 |>,
```

Reply:
thanks, fixed it